packages/cardano-services/src/DbSyncProvider.ts

```diff
@@ -3,19 +3,113 @@ import { Pool } from 'pg';
 
 const HEALTH_CHECK_QUERY = 'SELECT 1';
 
-export abstract class DbSyncProvider implements Provider {
-  public db: Pool;
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type AnyArgs = any[];
 
-  protected constructor(db: Pool) {
-    this.db = db;
+export const DbSyncProvider = <
```

Comment:
I was out of this refactoring topic and I need first to read the whole discussion above in order to understand each suggestion in detail, but at first look, my personal opinion is that having all these types and abstractions is a bit "overengineered". The reason is: it's initially hard for me to understand the usage of it without having the demo usage lines or any other readme doc

Also not sure how flexible for extensions would be, considering we need to add new functionality/deps to the base `DbSyncProvider` class (like `CardanoNode`)

Reply:
Do you like the original implementation in #428 better?

Reply:
Yep, so once I read all discussion and check the other PR will be able to express opinion, agree about the Mixin class and the need of docs/usage.

Also to keep in mind the new implementation should be opened for extensions easily, I refer to that `DbSyncProvider` will be probably extended in near future, even now with another dependency of `CardanoNode`

Reply:
The #428 approach looks much more intuitive and simplified, so my question is what was the key advantage of this #431 variant of implementation over the #428?

Reply:
Hi @IvayloAndonov , this implementation is more robust as we can trust the properties of _classes inheritance_, as an example this let us to check if a given `DbSyncProvider` is a `RunnableModule` as well through an `instanceof` check rather than checking by hands for the presence of methods.

Reply:
As agreed in the standup, I'm resolving this